src/array/schema.rs

```diff
@@ -20,41 +21,88 @@ impl ArrayType {
 
 /// Wrapper for the CAPI handle.
 /// Ensures that the CAPI structure is freed.
-pub(crate) struct RawSchema {
-    ffi: *mut ffi::tiledb_array_schema_t,
-}
-
-impl RawSchema {
-    pub fn new(ffi: *mut ffi::tiledb_array_schema_t) -> Self {
-        RawSchema { ffi }
-    }
+pub(crate) enum RawSchema {
+    Owned(*mut ffi::tiledb_array_schema_t),
 }
 
 impl Deref for RawSchema {
     type Target = *mut ffi::tiledb_array_schema_t;
 
     fn deref(&self) -> &Self::Target {
-        &self.ffi
+        let RawSchema::Owned(ref ffi) = *self;
+        ffi
     }
 }
 
 impl Drop for RawSchema {
     fn drop(&mut self) {
-        unsafe { ffi::tiledb_array_schema_free(&mut self.ffi) }
+        unsafe {
+            let RawSchema::Owned(ref mut ffi) = *self;
+            ffi::tiledb_array_schema_free(ffi)
+        }
     }
 }
 
 pub struct Schema<'ctx> {
     context: &'ctx Context,
     raw: RawSchema,
-    _domain: Domain<'ctx>,
+    domain: Domain<'ctx>,
 }
 
 impl<'ctx> Schema<'ctx> {
     pub(crate) fn as_mut_ptr(&self) -> *mut ffi::tiledb_array_schema_t {
         *self.raw
     }
 
+    pub fn domain(&self) -> &Domain<'ctx> {
+        &self.domain
+    }
+
+    /// Retrieve the schema of an array from storage
+    pub fn open(context: &'ctx Context, uri: &str) -> TileDBResult<Self> {
```

Comment:
The TileDB API's would call this public API `load` or similar. I would either go with that, or the same thing as for the other `load` calls I mentioned where we might want to be more rust-y and use either a `From` trait or a `from_uri` method.